service/src/main/kotlin/fi/espoo/evaka/assistanceneed/AssistanceNeedController.kt

```diff
@@ -54,7 +59,23 @@ class AssistanceNeedController(
     ): ResponseEntity<List<AssistanceNeed>> {
         Audit.ChildAssistanceNeedRead.log(targetId = childId)
         accessControl.requirePermissionFor(user, Action.Child.READ_ASSISTANCE_NEED, childId)
-        return assistanceNeedService.getAssistanceNeedsByChildId(db, childId).let(::ok)
+        return assistanceNeedService.getAssistanceNeedsByChildId(db, childId).filter { assistanceNeed ->
+            // If child is or has been in preschool, do not show pre preschool daycare assistance needs for non admin
+            if (!user.isAdmin) {
+                val preschoolPlacements = getChildPreschooPlacements(db, childId)
+                if (preschoolPlacements.any { preschoolPlacement ->
+                    assistanceNeed.startDate.isBefore(preschoolPlacement.startDate)
+                }
+                ) false else true
+            } else
+                true
+        }.let(::ok)
```

Comment:
If you remove `ResponseEntity<>` from return type, the `ok` wrapping should not be needed.